File: Advanced/PageAndRDLX/CustomTileProvider/C#/CustomTileProviders/BaseTileProvider.cs
using ActiveReports.Samples.CustomTileProviders.Properties;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ActiveReports.Samples.CustomTileProviders
{
	public abstract class BaseTileProvider
	{
		private string GetCopyright()
		{
			var titleNameKey = GetType().Name + "_TileProviderCopyright";
			var title = Resources.ResourceManager.GetString(titleNameKey);
			if (!string.IsNullOrEmpty(title) && title != titleNameKey)
			{
				return string.Format(title, DateTime.Now.Year);
			}

			return string.Empty;
		}

		public virtual string Copyright => GetCopyright();
	}
}

File: Advanced/PageAndRDLX/CustomTileProvider/VB.NET/CustomTileProviders/BaseTileProvider.vb
Public MustInherit Class BaseTileProvider
    Private Function GetCopyright() As String
        Dim titleNameKey = [GetType]().Name & "_TileProviderCopyright"
        Dim title = ActiveReports.Samples.CustomTileProviders.My.Resources.Resources.ResourceManager.GetString(titleNameKey)

        If Not String.IsNullOrEmpty(title) AndAlso title <> titleNameKey Then
            Return String.Format(title, DateTime.Now.Year)
        End If

        Return String.Empty
    End Function

    Public Overridable ReadOnly Property Copyright As String
        Get
            Return GetCopyright()
        End Get
    End Property
End Class

